test: 알람API 테스트 작성

JINSOLdev opened the pull request:
## 📝 요약
- 알람 API의 유닛 테스트를 작성했습니다.
- 기존 알람 SSE(서버 전송 이벤트) 로직은 interval(30000)이 컨트롤러 내부에 하드코딩되어 있었고, switchMap 연산자를 사용해서 마지막 이벤트만 방출하는 문제가 있었습니다. 이 구조에서는 타이머를 제어할 수 없어 테스트가 불안정했고, 매 tick마다 알람이 정상적으로 전송되는지 검증할 수 없었습니다.
- 그래서 주기 로직을 주입형 스트림(TICKER$)으로 분리하고 RxJS의 concatMap연산자를 사용해서 모든 tick을 순차적으로 처리하도록 했습니다.
- 따라서 테스트에서 타이머를 모킹하지 않고도 정확히 호출한 횟수만큼 알람 이벤트가 전송되는지 검증할 수 있었습니다.

## 📝 변경사항
1. `ticker.token.ts` 추가
- `export const TICKER$ = Symbol('TICKER$')`
- DI 토큰으로 알림 주기 신호 스트림을 주입 가능하게 만들었습니다.

2. `notifications.controller.ts` 수정
- interval( )의 하드코딩을 제거해서 테스트에서 제어 불가능했던 문제를 해결했습니다.
- switchMap > concatMap으로 변경하여 모든 tick을 순차적으로 처리해서 들어오는 숫차만큼 방출을 보장할 수 있습니다.
- `@Inject(TICKER$)` 주입 추가로 컨트롤러는 주기 로직에 직접 의존하지 않습니다.

3. `notifications.module.ts` 수정
- 운영 시에는 실제 30초 간격으로 알림을 전송하고, 테스트에서는 이 provider를 오버라이드해서 주기 제어를 가능하게 변경했습니다.

## 📝 추가설명
- 이번 작업의 목표는 주기 로직의 외부화와 테스트 안정성 확보였습니다.
- 기존에는 interval(30000)이 컨트롤러 내부에서 직접 생성되어서 테스트 시 실제로 30초를 기다리거나 fake timer를 사용해야 하는 불편함이 있었습니다. 이제는 TICKER$를 통해 주입형 Observable로 주기 신호를 제어할 수 있고, 테스트 환경에서는 of(1, 2)처럼 즉시 방출하는 스트림을 주입해서 시간에 영향을 받지 않는 예측 가능한 테스트가 가능해졌습니다. 그리고 concatMap을 도입해서 각 tick마다 알림이 빠짐없이 전송되도록 했고, FakeAuthGuard를 적용해 인증 의존성을 제거하여 테스트의 독립성을 강화했습니다.

## 🔗관련 이슈
- Closes #168 

## ✅ PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

## ✅ 테스트
<img width="657" height="320" alt="image" src="https://github.com/user-attachments/assets/b83a525d-036b-4c24-a83d-d87c0d949331" />
<img width="486" height="116" alt="image" src="https://github.com/user-attachments/assets/9f2fd360-7f47-4c61-b3af-4d8d55f784a6" />

JINSOLdev committed:
test: 알람API 테스트 작성

- ticker.token.ts : DI토큰 작성
- notification.controller.ts : @Inject(TICKER$) private readonly ticker$: Observable<number> 주입 추가, RxJS 파이프라인을 interval 직접 호출 > 주입 받은 ticker$ 사용, switchMap > concatMap 교체
- notifications.module.ts : 운영 환경 기본값으로 30초 주입

chya-chya reviewed (approved):
확인 완료했습니다!
함수명과 테스트제목이 직관적이라 리뷰하기 좋네요 :)

JINSOLdev committed:
fix: concatMap > switchMap

concatMap은 각 작업이 완료될땜까지 순차적으로 대기하며 큐에 쌓임. 
만약 DB조회가 ticker 간격보다 오래 걸리면, 요청들이 계속 큐에 쌓여 메모리 증가 및 알림 지연이 발생함. SSE는 과거 데이터보다 최신 데이터가 중요하므로 switchMap이 더 적절함. 따라서 switchMap으로 되돌림

JINSOLdev committed:
fix: 콜드 옵저버블로 인한 메모리 누수 및 다중 타이머 생성 문제 수정

interval(30_000)은 콜드 옵저버블을 생성하므로 이 토큰을 주입하는 각 컴포넌트마다 별도의 타이머가 생성되어 독립적으로 실행됨. 따라서 shareReplay(1) 연산자를 사용하여 핫 옵저버블로 만들어 모든 구독자가 단일 타이머를 공유하도록 수정함

jinseongnam reviewed (approved):
수고하셨습니다 확인했습니다
원래라면 제가 했어야하는 부분이였는데 도와주셔서 감사합니다
